app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
+    {
+        $formName       = $form->generateFormName();
+        $fields         = $form->getFields();
+        $autoFillFields = [];
+
+        /** @var Field $field */
+        foreach ($fields as $key => $field) {
+            if (in_array($field->getType(), ['date', 'datetime', 'hidden'])) {
+                $autoFillFields[$key] = $field;
+            }
+        }
+
+        foreach ($autoFillFields as $field) {
+            $fieldType = $field->getType();
+            $token     = $field->getDefaultValue();
```

Comment:
In Mautic we call tokens the variables in an email or landing page content like `{contactfield=firlstname}`. Can we use the name as the getter suggests to avoid confusion?
```suggestion
            $defaultValue     = $field->getDefaultValue();
```

Reply:
Even though they're not it curly braces, I saw the date strings as tokens, because they get replaced. But I get what you mean.